jobQueue/setup.js

```diff
@@ -18,19 +18,46 @@ const conf = require('../config');
 const featureToggles = require('feature-toggles');
 const urlParser = require('url');
 const kue = require('kue');
+const activityLogUtil = require('../utils/activityLog');
+
 const redisOptions = {
   redis: conf.redis.instanceUrl.queue,
 };
+
+const express = require('express');
+const server = require('http').Server(express());
+const SHUTDOWN_TIMEOUT = 5000;
+const SUCCESSFUL_EXIT = 0;
+
 const redisInfo = urlParser.parse(redisOptions.redis, true);
 if (redisInfo.protocol !== PROTOCOL_PREFIX) {
   redisOptions.redis = 'redis:' + redisOptions.redis;
 }
 
 const jobQueue = kue.createQueue(redisOptions);
 
+process.on('SIGTERM', () => {
+  const start = Date.now();
+  server.close(() => { // Stop http server and stop accepting any new request.
+    jobQueue.shutdown(SHUTDOWN_TIMEOUT, (err) => {
+      if (featureToggles.isFeatureEnabled('enableSigtermActivityLog')) {
+        const status = `Job queue shutdown: ${err || 'OK'}`;
```

Comment:
the value for this must be in double-quotes since there can be spaces and other special characters in it, otherwise log line parsing will not be happy

Reply:
I wrote a test to generate a weird callback err, but as I am running process.exit after the first test the test suit context (thread) was closed. 😄

Reply:
Will change it, thanks!